### DISCUSSION TITLE
Grouping test from makefile

### DISCUSSION THREAD

**Discussion by rtaylo45** [idaholab/moose#18756] [Q&A Tools] [answered]
Hello,

I was wondering if it is possible to group sets of test based on variables set in the Makefile? Ive used the `heavy` flag in the test file before, but wasn't sure if there was some sort of more complex system for grouping different test. For example, sometimes i might want to couple with another moose app that provides addition physics kernels, i would like to test the coupled capabilities and have the TestHarness system pickup that they need to be tested automatically. 

Thanks
Zack

**Comment by GiudGiud** [comment, marked as answer]
Hello

You could:
- add something to the test name or folder to tag it. Then have run_tests be ran as `--re <tag>` that will only run these tests
- use the `executable_pattern` option to restrict tests to a certain name for what executable runs the test
I did that here https://github.com/idaholab/virtual_test_bed/blob/main/mrad/steady/tests

Guillaume
